client/config/webpack.dev.js

```diff
@@ -0,0 +1,43 @@
+const webpack = require('webpack')
+const { merge } = require('webpack-merge')
+const common = require('./webpack.common.js')
+const paths = require('./paths')
+
+module.exports = merge(common, {
+  // Set the mode to development or production
+  mode: 'development',
+
+  // Control how source maps are generated
+  devtool: 'inline-source-map',
+
+  // Spin up a server for quick development
+  devServer: {
+    historyApiFallback: true,
+    contentBase: paths.build,
+    open: true,
+    compress: true,
+    hot: true,
+    port: 3000
+  },
+
+  module: {
+    rules: [
+      {
+        test: /\.s[ac]ss$/i,
+        use: [
+          // Creates `style` nodes from JS strings
+          "style-loader",
+          // Translates CSS into CommonJS
+          "css-loader",
+          // Compiles Sass to CSS
+          "sass-loader",
+        ],
+      },
+    ]
+  },
+
+  plugins: [
+    // Only update what has changed on hot reload
+    new webpack.HotModuleReplacementPlugin(),
```

Comment:
Have you thought about trying React Fast Refresh for hot module replacement?

Reply:
Wow, that's cool! I'll definitely try it out.